src/protocol/modulus_conversion/convert_shares.rs

```diff
@@ -108,6 +108,36 @@ impl ConvertShares {
     }
 }
 
+pub async fn convert_all_shares<F: Field>(
+    ctx: &ProtocolContext<'_, F>,
+    input: &[u64],
+) -> Result<Vec<Vec<Replicated<F>>>, BoxError> {
+    let converted_shares = try_join_all(zip(repeat(ctx), input).enumerate().map(
+        |(record_id, (ctx, row))| async move {
+            ConvertShares::new(XorShares {
+                num_bits: 64,
+                packed_bits: *row,
+            })
+            .execute(
+                ctx.narrow(&ModulusConversion(record_id.try_into().unwrap())),
+                RecordId::from(record_id),
+            )
+            .await
+        },
+    ))
+    .await?;
+
+    //Transpose the output to return bit-wise results
+    let cols = vec![Replicated::new(F::ZERO, F::ZERO); converted_shares.len()];
```

Comment:
I don't understand what's going on here. If we are sorting 1 million rows, then converted_shares.len() will take the value 1 million. How are there 1 million "columns"? I think this is an error.

Reply:
This is the number of columns which will be created in the transpose (which is same as number of rows = 1 million in the example). Let me make it more accessible. However, agree it would be better to not create extra memory for doing transpose. Working on it